src/pages/docs/concepts-reliability.mdx

```diff
@@ -0,0 +1,363 @@
+
+## Fault-Tolerance
+
+Golem's fault-tolerance guarantees are focused on ensuring that your workers
+are executed transactionally and that their state is durable, regardless of
+software or hardware faults, restarts, or updates.
+
+In order to more precisely define the scope of these guarantees, we need to
+introduce some terminology:
+
+- **Failure Event**. A _failure event_ is any event that causes a worker to
+  interrupt execution, but excluding all internal errors in the worker itself.
+  This includes operating system faults, hardware faults, disruptions
+  (termination through the container orchestrator), restarts, and updates.
+- **Supervision**. _Supervision_ is the process by which Golem detects failure
+  events and takes action to recover interrupted workers.
+- **Recovery**. _Recovery_ is the process by which Golem restores the state of
+  a worker to its state before the failure event, and resumes execution.
+
+### Limitations
+
+Golem's fault-tolerant guarantees are extensive, and they enable you to build
+highly reliable distributed systems without having to think about most failure
+scenarios. However, you should be aware of the following limitation:
+
+- **Execution Semantics**. For external systems (those not executed by Golem),
+  Golem generally guarantees only _at-least once semantics_ with respect to
+  the last external request made before the failure event.
+
+The following section discusses this limitation in more depth and what you can
+do to mitigate it.
+
+### Execution Semantics
+
+All local code, as well as all communication between workers, is executed with
+exactly-once semantics. All local code executed strictly inside Golem,
+including worker-to-worker communication, is guaranteed to have exactly-once
+semantics. This means that during recovery, the state of the worker is restored
+to the exact state it was in before the failure event, and no operations
+performed by the worker are repeated.
+
+However, for remote operations, such as invoking an HTTP API, Golem cannot
+generally guarantee exactly-once semantics. This is because a failure event
+may occur after a request has been sent, but before the response has been
+received. In this scenario, during the recovery, the request must be retried,
+which may lead to the remote system processing the same request multiple times.
+
+To mitigate this issue, you should consider using HTTP APIs that support the
+[The Idempotency-Key HTTP Header Field](https://datatracker.ietf.org/doc/draft-ietf-httpapi-idempotency-key-header/).
+This header allows you to provide a unique key for each request, which the
+remote system can use to ensure that the request is idempotent.
+
+In a future release, Golem will automatically generate idempotency keys for
+each request. For now, however, you can use the language-specific Golem SDK
+to generate an idempotency key, which you can then pass to the remote system.
+
+<Callout type="warning">
```

Comment:
This was not true but here is the fix https://github.com/golemcloud/golem/pull/773